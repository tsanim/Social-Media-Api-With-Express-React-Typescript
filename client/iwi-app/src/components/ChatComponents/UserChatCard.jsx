import React from 'react';
import URI from '../../config/config';
import PropTypes from 'prop-types';

function UserChatCard(props) {
    return (
        <figure className={ props.isMine ? "float-right" : null }>
            <img src={`${URI}/feed/image/${props.onlineUser.imageId}`} alt="userImg" />
            <div className="names">
                <figcaption>{props.onlineUser.username}</figcaption>
            </div>
        </figure>
    )
}

UserChatCard.propTypes = {
    onlineUser: PropTypes.object,
    isMine: PropTypes.bool
}


export default UserChatCard;