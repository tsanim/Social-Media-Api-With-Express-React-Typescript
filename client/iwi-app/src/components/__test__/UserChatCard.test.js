import React from 'react';

import { shallow } from 'enzyme';
import UserChatCard from '../ChatComponents/UserChatCard';

function userChatCardComponentSetup() {
    const props = {
        onlineUser: {},
        isMine: true
    }

    const enzymeWrapper = shallow(<UserChatCard {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('UserChatCard component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = userChatCardComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render figure tag with class "float-right"', () => {
        const { enzymeWrapper } = userChatCardComponentSetup();

        expect(enzymeWrapper.find('figure').hasClass('float-right')).toEqual(true);
    });

    it('should render img tag', () => {
        const { enzymeWrapper } = userChatCardComponentSetup();

        expect(enzymeWrapper.find('img')).toHaveLength(1);
    });

    it('should render div tag with class "names"', () => {
        const { enzymeWrapper } = userChatCardComponentSetup();

        expect(enzymeWrapper.find('div.names')).toHaveLength(1);
    });

    it('should render figcaption tag', () => {
        const { enzymeWrapper } = userChatCardComponentSetup();

        expect(enzymeWrapper.find('figcaption')).toHaveLength(1);
    });
});