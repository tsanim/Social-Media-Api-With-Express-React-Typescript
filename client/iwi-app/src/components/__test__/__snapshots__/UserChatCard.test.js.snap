// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChatCard component should render initial layout 1`] = `
Array [
  <figure
    className="float-right"
  >
    <img
      alt="userImg"
      src="http://localhost:9999/feed/image/undefined"
    />
    <div
      className="names"
    >
      <figcaption />
    </div>
  </figure>,
]
`;
